models/DataObject/Data/BlockElement.php

```diff
@@ -150,7 +148,17 @@ function ($currentValue) {
                     return $currentValue;
                 }
             ),
-            new UnmarshalMatcher()
+            new class(ElementInterface::class) extends TypeMatcher {
+                /**
+                 * @param mixed $element
+                 *
+                 * @return bool
+                 */
+                public function matches($element)
+                {
+                    return $element instanceof ElementInterface;
+                }
+            }
```

Comment:
Why do you need an anonymous class here? Shouldn't this be sufficient?

```suggestion
            new TypeMatcher(ElementInterface::class)
```